lib/SILOptimizer/IPO/ClosureSpecializer.cpp

```diff
@@ -1112,7 +1186,10 @@ bool SILClosureSpecializerTransform::gatherCallSites(
       // Uses may grow in this loop.
       for (size_t UseIndex = 0; UseIndex < Uses.size(); ++UseIndex) {
         auto *Use = Uses[UseIndex];
+        if (llvm::find(visitedUsers, Use) != visitedUsers.end())
```

Comment:
This is bad for efficiency and doesn't follow common patterns. The insert into visitedUsers itself will tell you if the insertion succeeds.

Reply:
Great, I'll use that. The reason for `visitedUsers` is because `MultipleClosureAI` gets confused later on. I was planning on trying to fix this pass so it would work with multiple closures. Maybe it would be better to land that patch first.